trustedentity/server/main.go

```diff
@@ -217,32 +214,34 @@ func CreatePKICertificate (
 
 // SecretKeeperServer is used to implement trustedentity.SecretKeeperServer.
 type secretKeeperServer struct {
-	x string
+	myVault *vaultapi.Logical
 }
 
-// SayMyReq implements trustedentity.SecretKeeperServer.SayMyReq service
-func (s *secretKeeperServer) SayMyReq(ctx context.Context, in *pb.MyReqRequest) (*pb.MyReqReply, error) {
+/*
+ GetVault -
+		the first caller initializes the vault and stores the reference to it in the struct
+		each subsequent caller verifies if the myVault element has been initialized
+			and if true, then just returns the myVault value
+
+		Returns - myVault value (a reference to the vault)
+*/
+func (s *secretKeeperServer) GetVault() (*vaultapi.Logical) {
```

Comment:
After giving some thought, it looks like this pattern will not work in concurrent environment. Shall I use singleton pattern instead? Do we have any examples in the code?